### ISSUE TITLE
SKTypeface.FromFamilyName does not change the font or text style in SkiaSharp.Views.Blazor

### ISSUE THREAD

**Issue by harveytriana** [mono/SkiaSharp#1902]
Unexpected behavior when changing the text font when drawing text. If you query what type of text it should have after assigning **Typeface**, it is always ```typeFace: Noto Mono.```

The following example illustrates the supposed issue,

```csharp
@page "/drawing-text"
@using SkiaSharp
@using SkiaSharp.Views.Blazor

<SKCanvasView OnPaintSurface="OnPaintSurface"  />

@code {
    void OnPaintSurface(SKPaintSurfaceEventArgs e)
    {
        var canvas = e.Surface.Canvas;

        var fontFamily = "Tahoma"; // "Arial" | "Calibri" | ... none works?
        var typeFace = SKTypeface.FromFamilyName(fontFamily, SKFontStyle.BoldItalic);
        // query
        Console.WriteLine("typeFace: {0}", typeFace.FamilyName); //? typeFace: Noto Mono

        using SKPaint paint = new() {
                Style = SKPaintStyle.Fill,
                TextSize = 26,
                IsAntialias = true,
                Typeface = typeFace
            };

        canvas.DrawText("Fancy Text", 30, 30, paint); //? draw with Noto Mono
    }
}
```
> The expected behavior is that the text is drawn with the text font and text style. I have tried the equivalent code in other scenarios and it works fine.

P.S. [Github of the example presented here](https://github.com/harveytriana/BlazorSkiSharpFontFamily)

**Comment by Gillibald** [comment]
Skia's font manager is highly limited under wasm so  you can't expect a functional font manager. It is better to load fonts explicitly via stream in a sandbox environment.

**Comment by harveytriana** [comment]
True Gillibald. Studying the «issue », it can be deduced that from WASM we do not have access to the Browser resources, so *SKTypeface* returns the *Default*, without warning. - Although I guess should be possible to do so, just like CSS you can use "Arial", to name something.

If anything, that's correct, I had already found a way to fix it.

- The file TTF file is added as an embedded resource (ideal)
- The font is loaded with SKTypeface.FromStream, reading the resource with Reflection.
- To improve the matter, it is provided as a Singleton service in the application. **Issue resolved**

**Comment by RChrisCoble** [comment]
This is a deal killer for using SkiaSharp on WASM if every font needs to be an embedded resource.  We have a drawing package we're porting to Blazor WASM with SkiaSharp wherein the user picks as many fonts as they want in the design portion, with the runtime portion showing the designed graphics.

@mattleibow do we have options here?

**Comment by Gillibald** [comment]
MAUI does the same there is no way around it and that's how the web works. Usually those assets are provided via a webservice so font assets can be cached by the browser you don't have to bloat up your complication size.

**Comment by Gillibald** [comment]
Yes exactly. Javascript fetch API is available via JSInterop or directly via Emscripten so you will get caching out of the box as long as your service provides proper headers.

**Comment by harveytriana** [comment]
In this post, https://www.blazorspread.net/blogview/82, I show possibly one of the best ways to handle fonts in Blazor, of course, it also covers MAUI
![image](https://user-images.githubusercontent.com/5245381/155205886-d38c4147-09ea-40b3-9e8e-ad9bf3986ead.png)

**Comment by mattleibow** [comment]
Can't you also hack the system and have a separate assembly with just fonts. And then use the feature of Blazor to load assemblies on demand? Is that even possible?

**Comment by harveytriana** [comment]
Yes

**Comment by jonchardy** [comment]
This is a major pain point. I work on a diagramming library and we're hoping to add Blazor as a target. We need to provide our users with flexibility in fonts. How would one go about creating an SKTypeface dynamically, from a given family name, where we don't control what potential families may be used?

**Comment by 18779758522** [comment]
> 在这篇文章中，https://www.blazorspread.net/blogview/82，我展示了可能是 Blazor 中处理字体的最佳方法之一，当然，它还涵盖了 MAUI ![图像](https://user-images.githubusercontent.com/5245381/155205886-d38c4147-09ea-40b3-9e8e-ad9bf3986ead.png)

cannot access

**Comment by mennowo** [comment]
Actually, the comment by @harveytriana [here](https://github.com/mono/SkiaSharp/issues/1902#issuecomment-1000528297) solves this issue. Embedding a font in my WASM Blazor app and loading it via a stream does indeed work.

A different solution seems unlikely, given that WASM has no direct access to browser resources, as far as I understand. Unless there is some clean way to do this via JS interop inside the SkiaSharp.Views.Blazor library.

In my opion, unless a potential better solution is doable some time soon, this issue should be closed for clarity.
